admin/services/pupil-history-service/data-access/pupil-history.data.service.ts

```diff
@@ -0,0 +1,236 @@
+import moment from 'moment';
+const sqlService = require('../../data-access/sql.service')
+const uuidValidate = require('uuid-validate')
+const R = require('ramda')
+
+export interface ICheckData {
+  id: number,
+  createdAt: moment.Moment
+  updatedAt: moment.Moment,
+  pupilId: number,
+  checkCode: string,
+  checkWindowId: number,
+  checkFormId: number,
+  pupilLoginDate: null | moment.Moment,
+  receivedByServerAt: null | moment.Moment,
+  isLiveCheck: boolean,
+  received: boolean,
+  complete: boolean,
+  completedAt: null | moment.Moment,
+  processingFailed: boolean,
+  createdByUserId: number,
+  inputAssistantAddedRetrospectively: boolean,
+  resultsSynchronised: boolean
+}
+
+export interface IPupilData {
+  id: number,
+  createdAt: moment.Moment,
+  updatedAt: moment.Moment,
+  schoolId: number,
+  foreName: string,
+  middleNames: string,
+  lastName: string,
+  gender: string,
+  dateOfBirth: moment.Moment,
+  upn: string,
+  urlSlug: string,
+  groupId: number | null,
+  currentCheckId: number | null,
+  checkComplete: boolean,
+  restartAvailable: boolean,
+  attendanceId: number | null,
+  foreNameAlias: string | null,
+  lastNameAlias: string | null
+}
+
+export interface ISchoolData {
```

Comment:
this kind of thing has become fragmented in `tslib`, so i would try and use more specific names like `pupilHistorySchoolData` if possible.  However, there are definitely generic types/models that might be useful to use globally.  School is definitely one of them

Reply:
Good point. The naming has been clarified.